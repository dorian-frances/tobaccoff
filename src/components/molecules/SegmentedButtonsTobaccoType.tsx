import { SegmentedButtons } from 'react-native-paper';
import { useState } from 'react';
import { StyleSheet } from 'react-native';

type SegmentedButtonsTobaccoTypeProps = {};

const SegmentedButtonsTobaccoType = ({}: SegmentedButtonsTobaccoTypeProps) => {
  const [value, setValue] = useState('industrielles');
  return (
    <SegmentedButtons
      value={value}
      onValueChange={setValue}
      buttons={[
        {
          value: 'industrielles',
          label: 'Industrielles',
          showSelectedCheck: true,
        },
        {
          value: 'roulées',
          label: 'Roulées',
          showSelectedCheck: true,
        },
      ]}
    />
  );
};

const styles = StyleSheet.create({
  container: {},
});

export default SegmentedButtonsTobaccoType;
