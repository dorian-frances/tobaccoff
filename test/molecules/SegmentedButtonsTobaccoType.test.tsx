import React from 'react';
import { fireEvent, render } from '@testing-library/react-native';
import SegmentedButtonsTobaccoType from '../../src/components/molecules/SegmentedButtonsTobaccoType';

describe('<SegmentedButtonsTobaccoType/>', () => {
  it('should render correctly', () => {
    const spyFunction = jest.fn();
    const component = render(
      <SegmentedButtonsTobaccoType
        defaultValue={'industrial'}
        getCigaretteType={spyFunction}
      />
    );
    expect(component.toJSON()).toMatchSnapshot();
  });

  it('should render with industrial as default type', () => {
    const spyFunction = jest.fn();
    const component = render(
      <SegmentedButtonsTobaccoType
        defaultValue={'industrial'}
        getCigaretteType={spyFunction}
      />
    );
    const industrialButton = component.getByTestId('industrialTypeButton');
    expect(industrialButton.props.accessibilityState.checked).toEqual(true);
  });

  it('should change value from industrial to rolled', () => {
    const spyFunction = jest.fn();
    const component = render(
      <SegmentedButtonsTobaccoType
        defaultValue={'industrial'}
        getCigaretteType={spyFunction}
      />
    );
    const industrialButton = component.getByTestId('industrialTypeButton');
    const rolledButton = component.getByTestId('rolledTypeButton');
    fireEvent.press(rolledButton);
    expect(rolledButton.props.accessibilityState.checked).toEqual(true);
    expect(industrialButton.props.accessibilityState.checked).toEqual(false);
  });
});
