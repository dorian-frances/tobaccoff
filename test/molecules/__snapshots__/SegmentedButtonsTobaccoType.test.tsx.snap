// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<SegmentedButtonsTobaccoType/> should render correctly 1`] = `
<View
  style={
    [
      {
        "flexDirection": "row",
      },
      undefined,
    ]
  }
>
  <View
    style={
      [
        {
          "backgroundColor": "rgba(232, 222, 248, 1)",
          "borderBottomRightRadius": 0,
          "borderColor": "rgba(121, 116, 126, 1)",
          "borderRadius": 20,
          "borderRightWidth": 0,
          "borderTopRightRadius": 0,
          "borderWidth": 1,
        },
        {
          "borderStyle": "solid",
          "flex": 1,
          "minWidth": 76,
        },
        [
          undefined,
          {},
        ],
      ]
    }
  >
    <View
      accessibilityRole="button"
      accessibilityState={
        {
          "busy": undefined,
          "checked": true,
          "disabled": false,
          "expanded": undefined,
          "selected": undefined,
        }
      }
      accessibilityValue={
        {
          "max": undefined,
          "min": undefined,
          "now": undefined,
          "text": undefined,
        }
      }
      accessible={true}
      collapsable={false}
      focusable={true}
      onBlur={[Function]}
      onClick={[Function]}
      onFocus={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        [
          {
            "overflow": "hidden",
          },
          false,
          {
            "borderBottomRightRadius": 0,
            "borderRadius": 20,
            "borderRightWidth": 0,
            "borderTopRightRadius": 0,
          },
        ]
      }
      testID="industrialTypeButton"
    >
      <View
        style={
          [
            {
              "alignItems": "center",
              "flexDirection": "row",
              "justifyContent": "center",
              "paddingHorizontal": 16,
              "paddingVertical": 9,
            },
            {
              "paddingVertical": 9,
            },
          ]
        }
      >
        <View
          collapsable={false}
          style={
            {
              "marginRight": 5,
              "transform": [
                {
                  "scale": 0,
                },
              ],
            }
          }
          testID="industrialTypeButton-check-icon"
        >
          <Text />
        </View>
        <Text
          numberOfLines={1}
          selectable={false}
          style={
            [
              {
                "color": "rgba(28, 27, 31, 1)",
                "fontFamily": "System",
                "fontSize": 14,
                "fontWeight": "500",
                "letterSpacing": 0.1,
                "lineHeight": 20,
              },
              {
                "textAlign": "left",
              },
              {
                "writingDirection": "ltr",
              },
              [
                {
                  "textAlign": "center",
                },
                {
                  "color": "rgba(28, 27, 31, 1)",
                  "fontFamily": "System",
                  "fontSize": 14,
                  "fontWeight": "500",
                  "letterSpacing": 0.1,
                  "lineHeight": 20,
                },
              ],
            ]
          }
        >
          Industrielles
        </Text>
      </View>
    </View>
  </View>
  <View
    style={
      [
        {
          "backgroundColor": "transparent",
          "borderBottomLeftRadius": 0,
          "borderColor": "rgba(121, 116, 126, 1)",
          "borderRadius": 20,
          "borderTopLeftRadius": 0,
          "borderWidth": 1,
        },
        {
          "borderStyle": "solid",
          "flex": 1,
          "minWidth": 76,
        },
        [
          undefined,
          {},
        ],
      ]
    }
  >
    <View
      accessibilityRole="button"
      accessibilityState={
        {
          "busy": undefined,
          "checked": false,
          "disabled": false,
          "expanded": undefined,
          "selected": undefined,
        }
      }
      accessibilityValue={
        {
          "max": undefined,
          "min": undefined,
          "now": undefined,
          "text": undefined,
        }
      }
      accessible={true}
      collapsable={false}
      focusable={true}
      onBlur={[Function]}
      onClick={[Function]}
      onFocus={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        [
          {
            "overflow": "hidden",
          },
          false,
          {
            "borderBottomLeftRadius": 0,
            "borderRadius": 20,
            "borderTopLeftRadius": 0,
          },
        ]
      }
      testID="rolledTypeButton"
    >
      <View
        style={
          [
            {
              "alignItems": "center",
              "flexDirection": "row",
              "justifyContent": "center",
              "paddingHorizontal": 16,
              "paddingVertical": 9,
            },
            {
              "paddingVertical": 9,
            },
          ]
        }
      >
        <Text
          numberOfLines={1}
          selectable={false}
          style={
            [
              {
                "color": "rgba(28, 27, 31, 1)",
                "fontFamily": "System",
                "fontSize": 14,
                "fontWeight": "500",
                "letterSpacing": 0.1,
                "lineHeight": 20,
              },
              {
                "textAlign": "left",
              },
              {
                "writingDirection": "ltr",
              },
              [
                {
                  "textAlign": "center",
                },
                {
                  "color": "rgba(28, 27, 31, 1)",
                  "fontFamily": "System",
                  "fontSize": 14,
                  "fontWeight": "500",
                  "letterSpacing": 0.1,
                  "lineHeight": 20,
                },
              ],
            ]
          }
        >
          Roul√©es
        </Text>
      </View>
    </View>
  </View>
</View>
`;
